src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,125 @@
-import React from 'react';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import classNames from 'classnames';
+import { Comment, CommentData } from '../types/Comment';
+import { createComment } from '../api/comments';
+import { warningTimer } from '../utils/warningTimer';
+
+type Props = {
+  postId: number | undefined;
+  setComments: Dispatch<SetStateAction<Comment[]>>
+  setIsHasError: Dispatch<SetStateAction<boolean>>;
+};
+
+export const NewCommentForm: React.FC<Props> = ({
+  postId,
+  setComments,
+  setIsHasError,
+}) => {
+  const [isLoading, setIsLoading] = useState(false);
+  const DefaultError = {
+    errorName: false,
+    errorEmail: false,
+    errorBody: false,
+  };
+  const nameError = 'errorName';
+  const emailError = 'errorEmail';
+  const bodyError = 'errorBody';
+  const [isError, setIsError] = useState(DefaultError);
+  const { errorName, errorBody, errorEmail } = isError;
+
+  const DefaultNewComment = {
+    postId: postId || 0,
+    name: '',
+    email: '',
+    body: '',
+  };
```

Comment:
1. When functional component is re-rendering - all variables and functions are creating again and again. So, we should move all constant values from component to avoid it's creating, so we won't waste CPU and memory. 
2. In your case it's better to create enum ErrorKeys instead of using 3 different variables.
3. In ideal you should move ErrorKeys, DefaultErrors, DefaultNewComment in separated files **(OPTIONAL)**
```suggestion

enum ErrorKeys {
  NameError = 'errorName',
  EmailError  = 'errorEmail',
  BodyError = 'errorBody'
}

// use computed property
const DefaultErrors = {
  [ErrorKeys.NameError]: false,   
  [ErrorKeys.EmailError]: false,
  [ErrorKeys.BodyError]: false,
};

 const DefaultNewComment = {
  postId: postId || 0,
  name: '',
  email: '',
  body: '',
};
  
export const NewCommentForm: React.FC<Props> = ({
  postId,
  setComments,
  setIsHasError,
}) => {
  const [isLoading, setIsLoading] = useState(false);

  const [isError, setIsError] = useState(DefaultError);
  const { errorName, errorBody, errorEmail } = isError;

```